# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/lib/wasm.js
# Commit: 85d47a33200c
# Full Hash: 85d47a33200c36a5f91f011f5713c34da79329ee
# Author: Julien Pages <jpages@mozilla.com>
# Date: 2025-09-15 21:44:14
# Regressor Bug: 1969509
# File Overlap Count: 7
# Overlapping Files: js/src/vm/ArrayBufferObject.cpp, js/src/wasm/WasmMemory.h, js/src/jit-test/tests/wasm/wasm-resizablearraybuffer-shared.js, js/src/jit-test/tests/wasm/memory64/basic.js, js/src/jit-test/tests/wasm/wasm-resizablearraybuffer.js
# Description:
#   Bug 1969509 - wasm: Update .maxByteLength to report the source wasm max pages in bytes. r=rhunt
#   
#   Following a spec change, 64-bit memory limits in wasm have been increased.
#   It's now possible for a wasm memory to have a limit bigger than the maximum
#   value of a 32-bits integer. To avoid an overflow, the maxByteLength getters
# ==============================================================================

diff -r 5860a981c750 -r 85d47a33200c js/src/jit-test/lib/wasm.js
--- a/js/src/jit-test/lib/wasm.js	Mon Sep 15 13:51:31 2025 +0000
+++ b/js/src/jit-test/lib/wasm.js	Mon Sep 15 14:01:27 2025 +0000
@@ -519,6 +519,12 @@
 // Max number of memories in a single wasm module.
 let MaxMemories = 100
 
+// Constants related to memory sizes.
+const MaxMemory64PagesValidation = BigInt(Math.pow(2, 37) - 1); // from spec
+const MaxTable64ElemsValidation = 0xFFFF_FFFF_FFFF_FFFFn; // from spec
+const MaxTableElemsRuntime = 10000000; // from WasmConstants.h
+const MaxUint32 = 0xFFFF_FFFF;
+
 // Common array utilities
 
 // iota(n,k) creates an Array of length n with values k..k+n-1